src/v2/api/services/DisciplineTeacherService.ts

```diff
@@ -123,14 +123,11 @@ export class DisciplineTeacherService {
       .roles.map((r) => r.role);
 
     const disciplineRoles = [];
```

Comment:
I think it`s better to change disciplineRoles type to Set, because, if you want put some logic after for cycle, you will work with "dirty" array.